docs/plugins/development/plugin-descriptor-file.asciidoc

```diff
@@ -0,0 +1,45 @@
+The plugin descriptor file is a Java properties file called
+ifeval::["{plugin-type}" ==  "analysis"]
+`stable-plugin-descriptor.properties`
+endif::[]
+ifeval::["{plugin-type}" == "other"]
+`plugin-descriptor.properties`
+endif::[]
+that describes the plugin. The file will be automatically created if you are 
+using {es}'s Gradle build system. If you're not using the gradle plugin, you 
+can create it manually using the following template.
+
+[source,properties]
+:plugin-properties-files: {elasticsearch-root}/build-tools/src/main/resources
+[source,yaml]
+----
+include::{plugin-properties-files}/plugin-descriptor.properties[]
+----
+
+[discrete]
+==== Mandatory elements for plugins
+
+
+[cols="<,<,<",options="header",]
+|=======================================================================
+|Element                    | Type   | Description
+
+|`description`              |String  | simple summary of the plugin
+
+|`version`                  |String  | plugin's version
+
+|`name`                     |String  | the plugin name
+ifeval::["{plugin-type}" ==  "analysis"]
+|`classname`                |String  | this property is for non-text analysis 
+plugins. Do not include this property.
+endif::[]
+ifeval::["{plugin-type}" == "other"]
+|`classname`                |String  | the name of the class to load, fully-qualified.
```

Comment:
There are a couple other possible properties here for non stable plugins: modulename and extended.plugins

Reply:
The table lists mandatory elements, and was copied from the existing docs. I'm happy to extend it to also include optional elements. We should then probably include `has.native.controller` and `licensed ` too?

Reply:
I would not include those. They are internal options really just for our modules.